ocaml/lambda/lambda.mli

```diff
@@ -279,6 +285,8 @@ and block_shape =
 and boxed_integer = Primitive.boxed_integer =
     Pnativeint | Pint32 | Pint64
 
+and unboxed_integer = boxed_integer
```

Comment:
maybe it's worth re-exporting the constructors here for more clarity, else this type declaration is a bit confusing.